nvflare/app_common/wf_comm/wf_comm_api.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import threading
+from typing import Callable, Dict, List, Optional, Union
+
+from nvflare.apis.controller_spec import SendOrder
+from nvflare.apis.fl_constant import ReturnCode
+from nvflare.app_common.abstract.fl_model import FLModel
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.wf_comm.wf_comm_api_spec import (
+    DATA,
+    MIN_RESPONSES,
+    RESP_MAX_WAIT_TIME,
+    RESULT,
+    SITE_NAMES,
+    STATUS,
+    TARGET_SITES,
+    TASK_NAME,
+    WFCommAPISpec,
+)
+from nvflare.fuel.data_event.data_bus import DataBus
+from nvflare.fuel.data_event.event_manager import EventManager
+from nvflare.private.defs import CommConstants
+
+
+class WFCommAPI(WFCommAPISpec):
```

Comment:
do we even need this class?
if we ask the users to just sub-class the WFController, that should be very similar to Holger's implementation of ModelController?

Reply:
this provides a better separation from nvflare specifics